solidity/ecdsa/contracts/libraries/EcdsaAuthorization.sol

```diff
@@ -15,25 +15,568 @@
+    ///         operator is known (`registerOperator` was called), the operator
+    ///         needs to update state of the sortition pool with a call to
+    ///         `joinSortitionPool` or `updateOperatorStatus`. After the
+    ///         sortition pool state is in sync, authorization decrease delay
+    ///         starts.
+    ///
+    ///         After authorization decrease delay passes, authorization
+    ///         decrease request needs to be approved with a call to
+    ///         `approveAuthorizationDecrease` function.
+    ///
+    ///         If there is a pending authorization decrease request, it is
+    ///         overwritten.
+    ///
+    /// @dev Should only be callable by T staking contract.
+    function authorizationDecreaseRequested(
+        Data storage self,
+        address stakingProvider,
+        uint96 fromAmount,
+        uint96 toAmount
+    ) internal {
+        require(
+            toAmount == 0 || toAmount >= self.parameters.minimumAuthorization,
+            "Authorization amount should be 0 or above the minimum"
+        );
+
+        address operator = self.stakingProviderToOperator[stakingProvider];
+
+        uint64 decreasingAt;
+
+        if (operator == address(0)) {
+            // Operator is not known. It means `registerOperator` was not
+            // called yet, and there is no chance the operator could
+            // call `joinSortitionPool`. We can let to approve authorization
+            // decrease immediately because that operator was never in the
+            // sortition pool.
+
+            // solhint-disable-next-line not-rely-on-time
+            decreasingAt = uint64(block.timestamp);
+        } else {
+            // Operator is known. It means that this operator is or was in
+            // the sortition pool. Before authorization decrease delay starts,
+            // the operator needs to update the state of the sortition pool
+            // with a call to `joinSortitionPool` or `updateOperatorStatus`.
+            // For now, we set `decreasingAt` as "never decreasing" and let
+            // it be updated by `joinSortitionPool` or `updateOperatorStatus`
+            // once we know the sortition pool is in sync.
+
+            // solhint-disable-next-line not-rely-on-time
+            decreasingAt = type(uint64).max;
+        }
+
+        uint96 decreasingBy = fromAmount - toAmount;
+
+        self.pendingDecreases[stakingProvider] = AuthorizationDecrease(
+            decreasingBy,
+            decreasingAt
+        );
+
+        emit AuthorizationDecreaseRequested(
+            stakingProvider,
+            operator,
+            fromAmount,
+            toAmount,
+            decreasingAt
+        );
+    }
+
+    /// @notice Approves the previously registered authorization decrease
+    ///         request. Reverts if authorization decrease delay have not passed
+    ///         yet or if the auhorization decrease was not requested for the
```

Comment:
```suggestion
    ///         yet or if the authorization decrease was not requested for the
```

Reply:
Ooops. Fixed in e80b3ff38b5970c9b4a9b73f59d93f497f797276.